app/controllers/scim_v2/users_controller.rb

```
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module ScimV2
+  class UsersController < Scimitar::ResourcesController
+    include BaseControllerActions
+
+    def create
+      super do |scim_resource|
+        storage_class.transaction do
+          user = storage_class.new
+          user.from_scim!(scim_hash: scim_resource.as_json)
+          user.firstname = "123" if user.firstname.blank?
+          user.lastname = "456" if user.lastname.blank?
```

:red_circle: Those "123" and "456" look like test code to me. Do we want to remove it?

As for the correct behaviour for when those required values are missing: I don't know. We could return an error response to the SCIM client, if we don't allow these to be empty, I guess?

---

It is an ugly way of dealing with a special case when SCIM Client does not send `firstname` and `lastname` to OpenProject. But for OP it is required to have these. Let me try to find more graceful placeholders for this scenario.

---

Correct me if I am wrong, but isn't this a "normal" error case?

I'd expect that we indicate first and last name as "required" in our schema. Thus a client that's not sending this information is sending something that's invalid according to our schema, so we can respond with HTTP 400 and indicate the `invalidValue` error.

See https://www.rfc-editor.org/rfc/rfc7644#section-3.12

Am I missing something in that regard?

---

Yes, your suggestion is absolutely valid and right. Let's go this way. Will be done in a follow-up.